lib/api-event-iterator.js

```diff
@@ -0,0 +1,17 @@
+import {on} from 'node:events';
+
+export async function * asyncEventIteratorFromApi(api) {
+	for await (const [plan] of on(api, 'run')) {
```

Comment:
is there a scenario where `api` will emit `run` multiple times? (other than watch mode maybe?)

Reply:
No, just in watch mode. May be helpful for the callback function to know that watch mode is active though.

Reply:
👍 might be good to add as a future enhancement